### ISSUE TITLE
Item metadata display shows old collection title after collection title was updated

### ISSUE THREAD

**Issue by andrewjbtw** [sul-dlss/argo#4653]
**Describe the bug**
[This collection](https://argo.stanford.edu/view/druid:ts561xq4138) title is: 
>Stanford Historical Society Oral History Program interviews, 1999-2013 (inclusive)

But items in the collection ([like this one](https://argo.stanford.edu/view/druid:sb753ms8796)) are showing an old title:
> Stanford Historical Society Oral History Program interviews, 1999-2022

I've tried reindexing the items but the title never changes. The incorrect title appears on at least the following pages:

- show page
- search results
- collection facet

**User Impact**
It's misleading because it looks like the collection metadata hasn't been updated and can lead people to not find the collection they're looking for.

**To Reproduce**
1. Go to an item [like](https://argo.stanford.edu/view/druid:sb753ms8796
2. Click on reindex
3. See no change in title

**Expected behavior**
Item titles should be updated when collection titles are updated.

**Comment by lwrubel** [comment]
I'm looking at this example druid now, and it shows `Stanford Historical Society Oral History Program interviews, 1999-2013 (inclusive)`.  So it must have eventually gotten the new display title. Some notes on where the info is coming from:

When I look at the [solr document for the item](https://argo.stanford.edu/view/druid:sb753ms8796.json), I see:
```
collection_title_ssim: [
"Stanford Historical Society Oral History Program interviews, 1999-2013 (inclusive)"
],
collection_title_tesim: [
"Stanford Historical Society Oral History Program interviews, 1999-2013 (inclusive)"
],
```
And the [collection's solr document](https://argo.stanford.edu/view/druid:ts561xq4138.json) shows:
```
obj_label_tesim: [
"Stanford Historical Society Oral History Program interviews, 1999-2012"
],
...
display_title_ss: "Stanford Historical Society Oral History Program interviews, 1999-2013 (inclusive)",
```

Based on the code in collection_concert, it appears the collection name text used in the record is coming from the solr document's `collection_title_ssim`.

Looks like there are still some items faceted with the old name though.

![Image](https://github.com/user-attachments/assets/dd7bf0d5-4caa-46fc-a5e1-5d1549408edd)

**Comment by andrewjbtw** [comment]
Yeah, I was going to come back and say that it seems to be slowly turning over to the new title.

**Comment by andrewjbtw** [comment]
I took this screenshot Tuesday and there hasn't been much change since then:
> <img width="1245" alt="Screenshot 2024-10-02 at 2 22 19 PM" src="https://github.com/user-attachments/assets/edad1c4e-7d18-46f7-b916-e94c6f336fca">

When I filed the issue, they all had the title ending in 2022.

**Comment by andrewjbtw** [comment]
All of these items now have the new, correct title.

New theory: rolling indexing eventually gets the new title assigned, but indexing that happens via workflows or using Argo's reindex does not.

**Comment by justinlittman** [comment]
@andrewjbtw Can this still be reproduced? Indexing has changed significantly since this ticket was created.

**Comment by andrewjbtw** [comment]
It happened again just before I left for vacation last month. By the time I returned, the collection titles had all eventually been updated to the new title. I haven't tried reproducing in stage yet.
